### PR TITLE
gex: 0.3.3 -> 0.3.8

### PR DESCRIPTION
**Pull request by Azd325**
###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by Azd325** [comment]
Result of `nixpkgs-review pr 225549` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>gex</li>
  </ul>
</details>

File: pkgs/applications/version-management/gex/default.nix [hunk 1]
```diff
@@ -2,25 +2,32 @@
 , stdenv
 , rustPlatform
 , fetchFromGitHub
+, pkg-config
+, openssl
+# libgit2-sys doesn't support libgit2 1.6 yet
```
Comment (edrex) [inline pkgs/applications/version-management/gex/default.nix:7]: Maybe change this to something like

```
# waiting on gex to update to libgit2-sys >= 0.15
```

(https://github.com/rust-lang/git2-rs/blob/41526ca62894cd68399148e3ec79c3e68c568f1e/libgit2-sys/CHANGELOG.md?plain=1#L31)

**Review by edrex — CHANGES_REQUESTED** [review]
Tested, seems to work. PR looks good other than comment. Tool is still not very usable due to lack of cursor features.

File: pkgs/applications/version-management/gex/default.nix (hunk 1 above)
Comment (Azd325) [reply pkgs/applications/version-management/gex/default.nix]: Sounds fine to me. Changed.

**Review by edrex — APPROVED** [review]

File: pkgs/applications/version-management/gex/default.nix [hunk 2]
```diff
@@ -2,25 +2,32 @@
...
     owner = "Piturnah";
     repo = pname;
     rev = "v${version}";
-    hash = "sha256-oUcQKpZqqb8wZDpdFfpxLpwdfQlokJE5bsoPwxh+JMM=";
+    hash = "sha256-pjyS0H25wdcexpzZ2vVzGTwDPzyvA9PDgzz81yLGTOY=";
   };
 
-  cargoHash = "sha256-ZFrIlNysjlXI8n78N2Hkff6gAplipxSQXUWG8HJq8fs=";
+  nativeBuildInputs = [ pkg-config ];
+  buildInputs = [ openssl libgit2_1_5 ];
```
Comment (drupol) [inline pkgs/applications/version-management/gex/default.nix:23]: Shouldn't libgit be in nativebuildinputs?

Comment (Azd325) [reply pkgs/applications/version-management/gex/default.nix:23]: How I understand it  relies on FFI, so we have to go this route.

https://discourse.nixos.org/t/failed-to-compile-openssl-sys-with-rust-on-macos/20785/2

https://johns.codes/blog/rust-enviorment-and-docker-build-with-nix-flakes

Comment (Piturnah) [reply pkgs/applications/version-management/gex/default.nix:23]: I believe git is needed in nativeBuildInputs as it is shelled out to in the [build script](https://github.com/Piturnah/gex/blob/70d59626ae41e6cebb3eeb8b949d2e8ac8823f77/build.rs#LL8C21-L8C21)

**Comment by Piturnah** [comment]
> Tested, seems to work. PR looks good other than comment. Tool is still not very usable due to lack of cursor features.

If you want to open an issue in gex about what cursor features you'd like me to prioritise I'd love to hear your feedback. Haven't had the time to spend on the project this year unfortunately

**Comment by drupol** [comment]
What's the status here? Is everything fixed?

File: pkgs/applications/version-management/gex/default.nix [hunk 3]
```diff
@@ -2,25 +2,32 @@
 , stdenv
 , rustPlatform
 , fetchFromGitHub
+, pkg-config
+, openssl
+# waiting on gex to update to libgit2-sys >= 0.15
```
Comment (SuperSandro2000) [inline pkgs/applications/version-management/gex/default.nix:7]: thats in the next release

**Comment by SuperSandro2000** [comment]
> What's the status here? Is everything fixed?

Someone pointed out that it is missing some feature which which is something for upstream.

**Pull request merged**